.github/workflows/test.yml

```diff
@@ -30,11 +29,48 @@ jobs:
         with:
           path: ./lib/
           key: ${{ runner.os }}-submodules
+      # Install Foundry
+      - name: install foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+      # Install Python
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - name: Load cached Poetry installation
+        id: cached-poetry
+        uses: actions/cache@v3
+        with:
+          path: ~/.local
+          key: poetry-${{ runner.os }}
+      - name: Install Poetry
+        if: steps.cached-poetry.outputs.cache-hit != 'true'
+        uses: snok/install-poetry@v1
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+          installer-parallel: true
+      - name: Load cached venv
+        id: cached-poetry-dependencies
+        uses: actions/cache@v3
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ hashFiles('**/poetry.lock') }}
+      - name: Setup the Kakarot submodule
+        run: make setup
+      # Lint
+      - name: Lint
+        run: |
+          cargo check &&
+          cargo fmt --all -- --check &&
+          cargo clippy --workspace --all-features --all-targets -- -D warnings
```

Comment:
Is there any reason for adding the linting here? I've had issues previously when jobs are tangled up like this in order to add a new job

Reply:
yeah, it's a bit annoying and I hope to make this better soon, but one needs the artifacts built locally to run clippy because clippy builds, and build needs the ABI .json files!

i've tried several ways to achieve this, but this one is the most straightforward, 
and we limit the number of times we build our CI